packages/api/src/modules/stats/stats.service.ts

```diff
@@ -12,6 +14,27 @@ class StatsService {
     async getRequestsPerMonthByYear(year: number, includesAdmin: boolean) {
         return await statsRepository.countRequestsPerMonthByYear(year, includesAdmin);
     }
+
+    async getMonthlyUserNbByYear(year: number) {
+        const start = firstDayOfPeriod(year);
+        let count = await userService.countBeforeDate(start);
+        const users = await userService.findAndSortByPeriod(start, nextDayAfterPeriod(year));
+        const monthlyCount = {};
+        let month = 0;
+        for (const user of users) {
+            if (!user) continue;
+            while (month != (user.signupAt as Date).getMonth()) {
```

Comment:
Y'a pas un problème ici ? Tu ne réinitialise jamais month dans ta boucle `user of users`. Du coup pour la première itération ça fonctionne mais la seconde pour le second `user` le mois ne sera pas à 0 ?

Ou alors je ne comprend pas l'algo !

Reply:
Les utilisateurs sont triés par ordre de date d'inscription croissant. Donc je compte tous ceux de janvier, puis tous ceux de février etc. C'est plus clair ? Je peux rajouter un commentaire s'il faut mais a priori je saurais pas trop quoi préciser

Reply:
ah ok donc ta boucle `while (month != (user.signupAt as Date).getMonth())` ne sert qu'à "passer" les mois sans utilisateurs ? Mais à ce moment là c'est redondant avec `if (!user) continue`.

Peut-être qu'un call sera plus rapide pour éclaircir ce point

Reply:
algo repris